delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2024 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.communitydelphi.api.ast.UnitImportNode;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFix;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFixEdit;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.UnitImportNameDeclaration;
+
+@Rule(key = "FullyQualifiedImport")
+public class FullyQualifiedImportCheck extends DelphiCheck {
+
+  @Override
+  public DelphiCheckContext visit(UnitImportNode unitImportNode, DelphiCheckContext context) {
+    if (!unitImportNode.isResolvedImport()) {
+      return context;
+    }
+
+    UnitImportNameDeclaration unitImportNameDeclaration = unitImportNode.getImportNameDeclaration();
+
+    String unitFullyQualifiedName = unitImportNameDeclaration.getOriginalDeclaration().getImage();
+    String unitImportName = unitImportNameDeclaration.getImage();
+
+    if (unitImportName.length() != unitFullyQualifiedName.length()) {
+      context
+          .newIssue()
+          .onNode(unitImportNode)
+          .withMessage(
+              "Specify the fully qualified unit name (found: \"%s\" expected: \"%s\").",
+              unitImportName, unitFullyQualifiedName)
+          .withQuickFixes(
+              QuickFix.newFix("Fix to \"%s\"", unitFullyQualifiedName)
+                  .withEdit(QuickFixEdit.replace(unitImportNode, unitFullyQualifiedName)))
```

Comment:
This would have the wrong effect when quick fixing unit imports with the `in` keyword:

```delphi
uses
  Unit in "My.Unit.pas";
```

would overwrite the `in "MyUnit.pas"` and become 

```delphi
uses
  My.Unit;
```

This should work:

```suggestion
                  .withEdit(QuickFixEdit.replace(unitImportNode.getNameNode(), unitFullyQualifiedName)))
```

Could you add a test for the `x in "y.pas"` case?

Reply:
Ok!

I added the test, but I'm having trouble making it work. I keep getting 'Unable to build Delphi File" due to the "Unit in 'UnitU.pas'"...

Reply:
If you copied the suggestion from @fourls directly, it's probably because you've got double quotes around the filename when it's expecting single quotes.

Reply:
I think the `in` syntax for the `uses` clause can only be used for a `program` file, not a `unit`.

I'm not sure if there is a `DelphiTestProgramBuilder` but if there is then that's what you want to use.

Reply:
https://github.com/integrated-application-development/sonar-delphi/blob/bb975dec84521054642a1431103de3ad4478d365/delphi-checks-testkit/src/main/java/au/com/integradev/delphi/builders/DelphiTestProgramBuilder.java

Reply:
Oops, ignore my accidental double quotes... 

Yes, as you say `in` does not work for normal units - `DelphiTestProgramBuilder` should work.

Reply:
@EduardoVaz06 did you see these responses a couple weeks ago?
Your test error should be pretty easy to fix with `DelphiTestProgramBuilder`.

Reply:
Hello! Sorry for the huge delay, I had some personal issues, but I just added the tests!

Reply:
I noticed that format check has failed, is there something else I need to do?

Reply:
Hi @EduardoVaz06, the project enforces `google-java-format` styling.
If you run `mvn install`, your code will automatically be reformatted correctly.

Reply:
Done, @Cirras!